File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/videos/VideosViewModel.java
package com.exact.twitch.ui.videos;


import androidx.lifecycle.MutableLiveData;
import androidx.annotation.Nullable;

import com.exact.twitch.model.video.Video;
import com.exact.twitch.repository.TwitchService;
import com.exact.twitch.ui.viewmodel.PagedListViewModel;

import javax.inject.Inject;

public class VideosViewModel extends PagedListViewModel<Video> {

    private MutableLiveData<CharSequence> currentOrderText = new MutableLiveData<>();
    private Integer selectedOrderItem;
    private String sort;

    private MutableLiveData<CharSequence> currentPeriodText;
    private Integer selectedPeriodItem;
    private String period;

    @Inject
    VideosViewModel(TwitchService repository) {
        super(repository);
    }

    public void loadVideos(@Nullable String game, @Nullable String period, @Nullable String broadcastTypes, @Nullable String language, @Nullable String sort, boolean reload) {
        loadData(repository.loadVideos(game, period, broadcastTypes, language, sort), reload);
    }

    public void loadFollowedVideos(String userToken, @Nullable String broadcastTypes, @Nullable String language, @Nullable String sort, boolean reload) {
        loadData(repository.loadFollowedVideos(userToken, broadcastTypes, language, sort), reload);
    }

    public void loadChannelVideos(Object channelId, @Nullable String broadcastTypes, @Nullable String sort, boolean reload) {
        loadData(repository.loadChannelVideos(channelId, broadcastTypes, sort), reload);
    }

    public MutableLiveData<CharSequence> getCurrentOrderText() {
        return currentOrderText;
    }

    public MutableLiveData<CharSequence> getCurrentPeriodText() {
        if (currentPeriodText == null) {
            currentPeriodText = new MutableLiveData<>();
        }
        return currentPeriodText;
    }

    public Integer getSelectedOrderItem() {
        return selectedOrderItem;
    }

    public void setSelectedOrderItem(Integer selectedOrderItem) {
        this.selectedOrderItem = selectedOrderItem;
    }

    public void setSelectedPeriodItem(Integer selectedPeriodItem) {
        this.selectedPeriodItem = selectedPeriodItem;
    }

    public Integer getSelectedPeriodItem() {
        return selectedPeriodItem;
    }

    public String getSort() {
        return sort;
    }

    public void setSort(String sort) {
        this.sort = sort;
    }

    public String getPeriod() {
        if (period == null) {
            period = "week";
        }
        return period;
    }

    public void setPeriod(String period) {
        this.period = period;
    }
}

File: app/src/main/java/com/exact/xtra/ui/videos/VideosViewModel.kt
package com.exact.xtra.ui.videos


import androidx.lifecycle.MutableLiveData
import com.exact.xtra.model.video.Video
import com.exact.xtra.repository.TwitchService
import com.exact.xtra.ui.Initializable
import com.exact.xtra.ui.common.PagedListViewModel
import javax.inject.Inject

class VideosViewModel @Inject constructor(
        private val repository: TwitchService) : PagedListViewModel<Video>(), Initializable {

    val sortText = MutableLiveData<CharSequence>()
    lateinit var sort: Sort

    val periodText by lazy { MutableLiveData<CharSequence>() }
    var period: Period? = null

    fun loadVideos(game: String? = null, broadcastType: BroadcastType = BroadcastType.ALL, language: String? = null, reload: Boolean) {
        loadData(repository.loadVideos(game, period ?: Period.WEEK, broadcastType, language, if (isInitialized()) sort else Sort.VIEWS, compositeDisposable), reload)
    }

    fun loadFollowedVideos(userToken: String, broadcastTypes: BroadcastType = BroadcastType.ALL, language: String? = null, reload: Boolean) {
        loadData(repository.loadFollowedVideos(userToken, broadcastTypes, language, if (isInitialized()) sort else Sort.VIEWS, compositeDisposable), reload)
    }

    fun loadChannelVideos(channelId: Any, broadcastTypes: BroadcastType = BroadcastType.ALL, reload: Boolean) {
        loadData(repository.loadChannelVideos(channelId, broadcastTypes, if (isInitialized()) sort else Sort.TIME, compositeDisposable), reload)
    }

    override fun isInitialized(): Boolean {
        return sortText.value != null && this::sort.isInitialized
    }
}
